File: Maven_22_06/src/main/java/exercicio1/banco/Banco.java
package exercicio1.banco;

import java.util.ArrayList;

public class Banco {
    private ArrayList<Cliente> clientes;

    public Banco() {
        // this.clientes = new Cliente[5]; - Codigo na versão com arrays
        clientes = new ArrayList<>();
    }

    public void adicionarCliente(Cliente cliente) {
        this.clientes.add(cliente);
    }

    public Cliente getCliente(int indice) {
        return this.clientes.get(indice);
    }

    public ArrayList<Cliente> getCliente(String nome, String sobrenome) {
        ArrayList<Cliente> result = new ArrayList<>();
        for (Cliente cliente: clientes) {
            if (cliente.getNome().equals(nome) && cliente.getSobrenome().equals(sobrenome)) {
                result.add(cliente);
            }
        }
        return result;
    }

    public int getNumeroDeClientes() {
        return clientes.size();
    }
}

File: Trabalho_JS/banco/dominio/Banco.js
export default class Banco {
    #clientes = new Array();
    
    constructor() {
        if (Banco.instance instanceof Banco) {
            return Banco.instance;
        }

        Object.freeze(this);
        Banco.instance = this;
    }

    getCliente(index) { return this.#clientes[index]; }

    filterCliente(nome, sobrenome) { 
        let result = new Array();
        for (let cliente of this.#clientes) {
            if (cliente.getNome().equals(nome) && cliente.getSobrenome().equals(sobrenome)) {
                result.push(cliente);
            }
        }
        return result;
    }

    getClientes() { return this.#clientes; }

    getNumeroClientes() { return this.#clientes.length; }

    adicionarCliente(cliente) { this.#clientes.push(cliente); }
}